### DISCUSSION TITLE
Why not connecting directly the accelerometer on the pi ?

### DISCUSSION THREAD

**Discussion by videogramme** [bassamanator/rp2040-zero-adxl345-klipper#2] [General]
Hello,
i'm wondering why cant i put the accelerometer directly on the pi without the pi zero ?
Thanks in advance

**Comment by bassamanator** [comment]
You can connect directly.

**Comment by videogramme** [comment]
Hello,
So why would i connect it by the raspberry zero ?
Thanks for your work

**Comment by bassamanator** [comment]
Also should mention the following:
- The rp2040-zero is not a raspberry pi zero.
- The benefits of using a rp2040-zero or a raspberry pi pico: it allows you to make use of the accelerometer more easily. You can just plug and play via a usb port, as opposed to having to fumble with gpio ports. For this same reason it's also a bit safer, it's hard to get it wrong, therefore you won't ever damage your host raspberry pi.
- The rp2040-zero (or raspberry pi pico) turns your acceloromter into a stand alone unit, that you can easily move between your other 3d printers.

Does that help?

You can find direct connection info on klipper docs, just search for `measuring resonances`.

**Comment by videogramme** [comment]
Hooo 
ok, i see. Thank you for the clear explanation.
Have a nice day
